demo-setups/fetchmail-compose.yaml

```diff
@@ -0,0 +1,58 @@
+# More information: https://github.com/orgs/docker-mailserver/discussions/3994#discussioncomment-9290570
+
+services:
+  dms-fetch:
+    image: ghcr.io/docker-mailserver/docker-mailserver:latest
+    hostname: mail.example.test
+    environment:
+      ENABLE_FETCHMAIL: 1
+      # We change this setting to 10 for quicker testing:
+      FETCHMAIL_POLL: 10
+    # Link the DNS lookup `remote.test` to resolve to the `dms-remote` container IP (for `@remote.test` address):
+    # This is only for this example, since no real DNS service is configured, this is a Docker internal DNS feature:
+    links:
+      - "dms-remote:remote.test"
+    # NOTE: Optional, You only need to publish ports if you want to verify via your own mail client.
+    #ports:
+    #  - "465:465"  # ESMTP (implicit TLS)
+    #  - "993:993"  # IMAP4 (implicit TLS)
+    # You'd normally use volumes here, for simplicity of the example, all config is contained within `compose.yaml`:
+    configs:
+      - source: dms-accounts-fetch
+        target: /tmp/docker-mailserver/postfix-accounts.cf
+      - source: fetchmail
+        target: /tmp/docker-mailserver/fetchmail.cf
```

Comment:
For reference, if anyone does expand on this in future IIRC, the following info might be worth being aware of.

---

It's possible to split these out to separate `compose.yaml` files and use the Docker Compose feature [`include`](https://docs.docker.com/compose/compose-file/14-include/) or the [merge/override feature](https://docs.docker.com/compose/compose-file/13-merge/#sequence) to share common configs.

Notably, you will have trouble merging YAML lists/sequences, which I think is only supported by multiple `-f` overrides (_requires the same service name_), whereas for YAML maps you can use [Fragments / Extensions](https://docs.docker.com/compose/compose-file/10-fragments/) (_YAML anchors/aliases + YAML merge `<<` syntax_).

- That can make `environment` more suited to map style instead of list, since that supports both forms.
- One notable difference over `volumes` with `configs` is that a bind mount volume will persist changes to disk whereas the `configs` approach here is useful for immutable init state when creating the container, but it can be modified after at runtime.
  - Re-creating the container will start with the initial config in `compose.yaml` again. This is the main value besides self-containing the example into a single file.
  - It might be possible to get the same with `volumes` if some of the volume driver has a setting to emulate one-way sync (_`:ro` / read-only does but it prevents writes in the container_)..

If there are separate YAML snippets to be composed, you just run [`docker compose config`](https://docs.docker.com/reference/cli/docker/compose/config/) instead of the equivalent `docker compose up` to output the full `compose.yaml`. If examples grow it may be useful to know about that :)